ksp_plugin_adapter/loader.cs

```diff
@@ -48,6 +48,13 @@ internal static class Loader {
              string.Join("', '", possible_dll_paths) + "' in directory '" +
              Directory.GetCurrentDirectory() + "'.";
     }
+    foreach (char c in Directory.GetCurrentDirectory()) {
+      if (c >= 128) {
+        return Directory.GetCurrentDirectory() +
+               " contains the non-ASCII character " + c +
+               "; this is known to confuse Mono.";
+      }
+    }
```

Comment:
The bug could be Windows-only and is probably system-dependent (for instance, on recent versions of Windows, the code page can be 65001, in which case interpreting UTF-8 bytes in the current code page woould work).

Let’s flag this potential issue only if an error occurs, as we do with `is_cxx_installed`, instead of making it prevent us from even trying.

Reply:
#2554.